lonestar/analytics/cpu/independentset/independent_set_cli.cpp

```diff
@@ -0,0 +1,119 @@
+#include "katana/ParallelSTL.h"
+#include "katana/Reduction.h"
+#include "katana/Timer.h"
+#include "katana/analytics/independent_set/independent_set.h"
+
+namespace {
+
+using namespace katana::analytics;
+
+const char* name = "Maximal Independent Set";
+const char* desc =
+    "Computes a maximal independent set (not maximum) of nodes in a graph";
+const char* url = "independent_set";
+
+namespace cll = llvm::cl;
+static cll::opt<std::string> inputFile(
+    cll::Positional, cll::desc("<input file>"), cll::Required);
+
+static cll::opt<IndependentSetPlan::Algorithm> algo(
+    "algo", cll::desc("Choose an algorithm:"),
+    cll::values(
+        clEnumValN(IndependentSetPlan::kSerial, "Serial", "Serial"),
+        clEnumValN(
+            IndependentSetPlan::kPull, "Pull",
+            "Pull-based (node 0 is initially in the independent set)"),
+        //        clEnumValN(
+        //            IndependentSetPlan::kNondeterministic, "Nondeterministic",
+        //            "Non-deterministic, use bulk synchronous worklist"),
+        //        clEnumValN(
+        //            IndependentSetPlan::kDeterministicBase, "DeterministicBase",
+        //            "use deterministic worklist"),
+        clEnumValN(
+            IndependentSetPlan::kPriority, "Priority",
+            "prio algo based on Martin's GPU ECL-MIS algorithm (default)"),
+        clEnumValN(
+            IndependentSetPlan::kEdgeTiledPriority, "EdgeTiledPriority",
+            "edge-tiled prio algo based on Martin's GPU ECL-MIS algorithm")),
+    cll::init(IndependentSetPlan::kPriority));
+
+}  // namespace
+
+int
+main(int argc, char** argv) {
+  std::unique_ptr<katana::SharedMemSys> G =
+      LonestarStart(argc, argv, name, desc, url, &inputFile);
+
+  katana::StatTimer totalTime("TimerTotal");
+  totalTime.start();
+
+  if (!symmetricGraph) {
+    KATANA_DIE(
+        "independent set requires a symmetric graph input;"
+        " please use the -symmetricGraph flag "
+        " to indicate the input is a symmetric graph");
+  }
+
+  std::cout << "Reading from file: " << inputFile << "\n";
+  std::unique_ptr<katana::PropertyFileGraph> pfg =
+      MakeFileGraph(inputFile, edge_property_name);
+
+  std::cout << "Read " << pfg->num_nodes() << " nodes, " << pfg->num_edges()
+            << " edges\n";
+
+  IndependentSetPlan plan = IndependentSetPlan::FromAlgorithm(algo);
+
+  if (auto r = IndependentSet(pfg.get(), "indicator", plan); !r) {
+    KATANA_LOG_FATAL("Failed to run algorithm: {}", r.error());
+  }
+
+  auto stats_result = IndependentSetStatistics::Compute(pfg.get(), "indicator");
+  if (!stats_result) {
+    KATANA_LOG_FATAL("Failed to compute statistics: {}", stats_result.error());
+  }
+  auto stats = stats_result.value();
+  stats.Print();
+
+  if (!skipVerify) {
+    if (IndependentSetAssertValid(pfg.get(), "indicator")) {
+      std::cout << "Verification successful.\n";
+    } else {
+      KATANA_LOG_FATAL("verification failed");
+    }
+  }
```

Comment:
What about output?

Reply:
The original didn't produce an output file. I could add it I guess. But I was just trying to duplicate the original.

Reply:
I was trying to put one even if it is not there. If it is easy to todo, I would say let's do it? It should only be in the cli.cpp, right?